pkg/scheduler/core/generic_scheduler.go

```diff
@@ -188,3 +194,420 @@ func (g *genericScheduler) chooseSpreadGroup(spreadGroup *util.SpreadGroup) []st
 	}
 	return feasibleClusters
 }
+
+func (g *genericScheduler) assignReplicas(clusters []*clusterv1alpha1.Cluster, replicaSchedulingStrategy *policyv1alpha1.ReplicaSchedulingStrategy, object *workv1alpha1.ObjectReference) ([]workv1alpha1.TargetCluster, error) {
+	if len(clusters) == 0 {
+		return nil, fmt.Errorf("no clusters available to schedule")
+	}
+	if object.Replicas == 0 {
+		targetClusters := make([]workv1alpha1.TargetCluster, len(clusters))
+		for i, cluster := range clusters {
+			targetClusters[i] = workv1alpha1.TargetCluster{Name: cluster.Name}
+		}
+		return targetClusters, nil
+	}
+	if replicaSchedulingStrategy != nil && replicaSchedulingStrategy.WeightPreferences != nil {
+		return g.calculateReplicasWithWight(clusters, replicaSchedulingStrategy.WeightPreferences.StaticWeightList, object.Replicas)
+	}
+	if replicaSchedulingStrategy != nil {
+		if replicaSchedulingStrategy.ResourcesPreferences != nil {
+			return g.calculateReplicasWithResources(clusters, replicaSchedulingStrategy.ResourcesPreferences, object)
+		}
+		if replicaSchedulingStrategy.WeightPreferences != nil {
```

Comment:
Hi, it seems like the code here will never be executed since you have the condition "if replicaSchedulingStrategy != nil && replicaSchedulingStrategy.WeightPreferences != nil"  above.

Reply:
The code above is redundant。I will delete it.